### ISSUE TITLE
Add autoconfig for QPid

### ISSUE THREAD

**Issue by dsyer** [spring-projects/spring-boot#7018]
Apparently QPid has AMQP 0.9.1 support now, so it works with Spring AMQP. Here's some working code: https://github.com/spring-guides/gs-messaging-rabbitmq/pull/12.

**Comment by garyrussell** [comment]
There are some limitations [listed here](https://www.rabbitmq.com/interoperability.html) - scroll down to "RabbitMQ clients » Qpid Java broker".

I believe the reference to basic.qos is the [difference with the 'global' flag mentioned here](http://www.rabbitmq.com/consumer-prefetch.html) which shouldn't matter because we only have one consumer per channel.

We also support some rabbitmq extensions (such as the delayed message exchange plugin).

On the face of it, it doesn't look like they are significant enough to have to fork spring-rabbit to spring-qpid091, we might just have to document the limitations.

**Comment by dsyer** [comment]
Sounds good to me. Good enough to warrant the autoconfig in Spring Boot anyway. The limitations should be documented in Spring AMQP I guess.

**Comment by garyrussell** [comment]
Some observations after trying to run some of the Spring AMQP integration test suite:

1. Spring AMQP will need to tweak its `BrokerRunning` JUnit `@Rule` - RabbitMQ returns `deleteOk` when deleting a non-existent queue, QPID closes the channel with a  queue not found exception, causing the rule to fail.

1. I see some problems, with template operations, mostly with sendAndReceive operations - for example the reply doesn't show up even though the temporary queue appears in the broker, has a consumer, and the reply message, but the consumer doesn't get the message for some reason. Of course, RabbitMQ's fast (direct) replyTo is not available, so we fall back to using a temporary queue (unless a fixed replyTo queue is configured).

1. Listener container tests seem better but I still see some failures.

I have only tried to run a few test cases so far.

Bottom line, I think, for us to officially support this, we'll need to invest some effort into tracking down these issues.

cc/ @artembilan

**Comment by philwebb** [comment]
We're cleaning out the issue tracker and closing issues that we've not seen much demand to fix. Feel free to comment with additional justifications if you feel that this one should not have been closed.
